learningresources/tests/test_models.py

```diff
@@ -190,3 +193,40 @@ def test_preview_url(self):
             resource.url_name = url_name
             url = get_preview_url(**kwargs)
             self.assertEqual(url, wanted)
+
+    def test_index_update_on_save(self):
+        """
+        Test that creating or saving a learning resource updates the index.
+        """
+        resource1 = LearningResource.objects.create(
+            course=self.resource.course,
+            learning_resource_type=self.resource.learning_resource_type
+        )
+
+        # New resource should show up in search results.
+        search_url = "{repo_base}{repo_slug}/search/".format(
+            repo_base=REPO_BASE,
+            repo_slug=self.repo.slug
+        )
+        resp = self.client.get(search_url)
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+        results = json.loads(resp.content.decode('utf-8'))['results']
+        results = [
+            result for result in results
+            if result['id'] == resource1.id
+        ]
+        # We only index created resources on import so we
+        # shouldn't have an index update here.
+        self.assertEqual(len(results), 0)
+
+        # Resource change should show up in search results.
+        resource1.description = "new description here"
+        resource1.save()
+        resp = self.client.get(search_url)
+        self.assertEqual(resp.status_code, HTTP_200_OK)
+        results = json.loads(resp.content.decode('utf-8'))['results']
+        results = [
```

Comment:
This bit might be clearer using a nested function:

```python
def resource_in_results(resource, results):
    for result in results:
        if result['description'] == resource.description
        return True
    return False
```

self.assertTrue(resource_in_results(resource1, results))